packages/eventual-send/src/index.js

```diff
@@ -473,7 +474,8 @@ export function makeHandledPromise(Promise) {
     // We return a handled promise with the default unfulfilled handler.
     // This prevents a race between the above Promise.resolves and
     // pipelining.
-    return new HandledPromise(executor);
+    returnedP = new HandledPromise(executor);
```

Comment:
You should get rid of the `let` on line 442 and make this be
```js
const returnedP = ...
```
You'll also need to suppress the eslint use-before-define warning. This is the right thing all the way around.
   * Forcing outselves to insert a comment to suppress the warning flags that something unusual is going on.
   * The temporal dead zone of the `const` declaration is your friend. If the variable is read before its defined, that would be a serious bug. The early `let` and late initialization would mask the bug with an `undefined`. Better to get a noisy throw.